pkg/proxy/ipvs/proxier.go

```diff
@@ -2069,6 +2069,12 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 				newEndpoints = localReadyTerminatingEndpoints
 			}
 		}
+
+		// Insert non-local endpoints to mimic the behavior of the iptables proxier
+		// https://github.com/kubernetes/kubernetes/issues/93456
```

Comment:
The comment is incorrect; this does not mimic the behavior of the iptables proxier.

The correct implementation of proxying requires that the proxy is able to have one rule for delivering internally-originated traffic (ie, from a pod or node), and a different rule for delivering externally-originated traffic. With the graduation of Internal Traffic Policy and Topology-Aware Hints, there are potentially cases where on a given node, internal traffic must be delivered to one set of pods, and external traffic must be delivered to a completely disjoint set of pods.

It seems like this hack will make things _better_ for IPVS users, so it might be good to merge it, but it's not correct.

(The patch will have no ill effects on `externalTrafficPolicy: Local` services when everything is working correctly, but it means they won't fail correctly when things break; if a loadbalancer sends to a node with no endpoint [due to race conditions], the required behavior is that the node drop the packet. But with this patch, it will accept it instead, and then lose the source IP.)

Reply:
> But with this patch, it will accept it instead, and then lose the source IP.

It will not be dropped, but the connect will not be accepted, please see https://github.com/kubernetes/kubernetes/issues/93456#issuecomment-733069629

Reply:
ok, well, that basically means it drops the connection in a roundabout and inefficient way. :slightly_smiling_face: 
So even better.